tests/test_commands.py

```diff
@@ -550,13 +551,95 @@ def test_url(self, url="test.com", domain="test.com"):
+            ).group(1),
+        )
+
+    def test_https_url_schema_templates(self):
+        self.test_start_urls_schema(
+            template="basic", url="https://test.com", schema="https"
+        )
+        self.test_start_urls_schema(
+            template="crawl", url="https://test.com", schema="https"
+        )
+        self.test_start_urls_schema(
+            template="xmlfeed", url="https://test.com", schema="https"
+        )
+        self.test_start_urls_schema(
+            template="csvfeed", url="https://test.com", schema="https"
+        )
+
+    def test_http_url_schema_templates(self):
+        self.test_start_urls_schema(
+            template="basic", url="http://test.com", schema="http"
+        )
+        self.test_start_urls_schema(
+            template="crawl", url="http://test.com", schema="http"
+        )
+        self.test_start_urls_schema(
+            template="xmlfeed", url="http://test.com", schema="http"
+        )
+        self.test_start_urls_schema(
+            template="csvfeed", url="http://test.com", schema="http"
+        )
+
+    def test_start_urls_matches_input(self, template="basic", url="https://test.com"):
+        self.assertEqual(
+            0, self.call("genspider", "--force", "-t", template, "test_name", url)
+        )
+        self.assertEqual(
+            url,
+            self.find_in_file(
+                Path(self.proj_mod_path, "spiders", "test_name.py"),
+                r'start_urls\s*=\s*\[[\'"](.+)[\'"]\]',
+            ).group(1),
+        )
+
+    def test_start_urls_substitution_with_schema(self):
+        self.test_start_urls_matches_input(
+            template="basic", url="https://test.com/full/path"
+        )
+        self.test_start_urls_matches_input(
+            template="xmlfeed", url="https://test.com/full/path"
+        )
+        self.test_start_urls_matches_input(
+            template="crawl", url="https://test.com/full/path"
+        )
+        self.test_start_urls_matches_input(
+            template="csvfeed", url="https://test.com/full/path"
+        )
+
+    def test_start_urls_no_schema(self, template="basic", url="test.com"):
+        self.assertEqual(
+            0, self.call("genspider", "--force", "-t", template, "test_name", url)
+        )
+        domain = extract_domain(url)
+        self.assertEqual(
+            domain,
+            self.find_in_file(
+                Path(self.proj_mod_path, "spiders", "test_name.py"),
+                r'start_urls\s*=\s*\[[\'"]http://(.+)/.*?[\'"]\]',
+            ).group(1),
+        )
+
+    def test_start_urls_substitution_no_schema(self):
+        self.test_start_urls_no_schema(template="basic", url="example.com")
+        self.test_start_urls_no_schema(template="crawl", url="example.com")
+        self.test_start_urls_no_schema(template="xmlfeed", url="example.com/full/path")
+        self.test_start_urls_no_schema(template="csvfeed", url="example.com/full/path")
```

Comment:
Interesting case. I wonder if we should support it differently. But it is a rare case, and the solution is simple (use a proper URL!), so it might not be worth the trouble.

Reply:
I actually didn't mean to leave the `full/path` part on those.  I don't think it's necessary to handle differently.  

I am going to remove the path extensions.